backend/registration-service/registration-service-chart/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ include "registration-service.fullname" . }}
+  labels:
+    {{- include "registration-service.labels" . | nindent 4 }}
+spec:
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: {{ .Release.Namespace }}
+  egress:
+    - {}
+    # - to:
+    #   {{- range .Values.namespaceToAllowEgress }}
+    #   - namespaceSelector:
+    #       matchLabels:
+    #         name: {{ . }}
+    #   {{- end }}
+    #   {{- range .Values.ipBlockToAllowEgress }}
+    #   - ipBlock:
+    #       cidr: {{ . }}
+    #  {{- end }}
```

Comment:
@ezramorris I've been looking into this one and it'd be great to get your opinion.


Completely open egress is not what we want here, we want to limit egress to just other namespaces (where an ActiveGate would run) and other IP addresses (the various managed services like RDS). If we put the IP addresses in here then they become 'public', which we may not be happy with.

I thought I could pull these somehow from a Kubernetes secret but I really need these at templating time, so I think a Kubernetes secret is out of the question.

Options that I can see are:

1. Have an open egress network policy in code. Make a comment that it isn't best practice but it saves revealing IPs
2. Check the IPs in to Git
3. Some other option I haven't considered where they can be somehow masked

Reply:
This has now been resolved, the IP addresses for the networkpolicy can be passed using the command line flags run with Helm. The IP addresses themselves will be committed as secrets to run in the Drone pipeline.